src/action/csfle/encrypt.rs

```diff
@@ -142,12 +148,10 @@ pub struct RangeOptions {
     pub precision: Option<i32>,
 }
 
-impl<Mode> Encrypt<'_, Mode> {
-    option_setters!(options: EncryptOptions;
-        contention_factor: i64,
-        range_options: RangeOptions,
-    );
-}
+#[option_setters(EncryptOptions, skip = [query_type])]
```

Comment:
Here's one of the fun corner cases!  The `Encrypt` action is shared across both `encrypt` and `encrypt_expression`; we want to expose the `query_type` option for the former, both in code and documentation, but not the latter.  So:
* For the generic `impl` block, the generated setter for `query_type` is skipped, and it's manually implemented for `Encrypt<'_, Value>` only.
* For the generated documentation for `encrypt`, the manually written setter for `query_type` is added to the list
* For the generated documentation for `encrypt_expr`, it isn't

Reply:
Somewhat tangential, but this led me to find a niche corner case that can lead to bad behavior when using `with_options` in a call to `encrypt_expression`:

```
let options = EncryptOptions::builder()
    .contention_factor(0)
    .range_options(range_options)
    .build();
let result = client_encryption
    .encrypt_expression(...)
    .with_options(options)
    .await?;
```

The call to `encrypt_expression` sets the `query_type` to the correct value on the options [internally](https://docs.rs/mongodb/latest/src/mongodb/action/csfle/encrypt.rs.html#60), but chaining `with_options` wipes this value, and the following error is returned: `EncryptExpression may only be used for range queries`.

We should add the following validation to the `IntoFuture` impl for `Encrypt<Expression>`:
- if `query_type` is set to `Some(<string not equal to "range">)`, return an error indicating that `query_type` cannot be set via `with_options` for calls to `encrypt_expression`
- if `query_type` is set to `None`, manually re-set the value to `Some("range")` as the user inadvertently overrode it by the call to `with_options`

I can make a follow-up PR to fix this; we're also missing a `TypedBuilder` impl for `EncryptOptions` so I'll add that too.